### PR TITLE
get_result: introduce kz_datamgr:get_result/3

### PR DESCRIPTION
**Pull request by fenollp**
Places to use this were found using

```shell
git grep -InE 'ok.+\[\].+error'
```

### CODE REVIEW

File: core/kazoo_data/src/kz_datamgr.erl [hunk 1]
```diff
@@ -1075,10 +1076,18 @@ get_result_keys(DbName, DesignDoc, Options) ->
...
-
--spec get_keys(kz_json:object()) -> kz_json:key().
-get_keys(JObj) -> kz_json:get_value(<<"key">>, JObj).
+    [kz_json:get_value(<<"key">>, JObj)
+     || JObj <- JObjs
+    ].
+
+-spec get_result(ne_binary(), ne_binary(), view_options()) -> {'ok', kz_json:object()} |
+                                                              data_error().
+get_result(DbName, DesignDoc, Options) ->
```
Comment (lazedo) [inline core/kazoo_data/src/kz_datamgr.erl]: can we change this to `get_single_result` instead ?

File: core/kazoo_data/src/kz_datamgr.erl [hunk 2]
```diff
@@ -1075,10 +1076,18 @@ get_result_keys(DbName, DesignDoc, Options) ->
...
+    case kz_datamgr:get_results(DbName, DesignDoc, Options) of
+        {'ok', []} -> {'error', 'not_found'};
+        {'ok', [Result]} -> {'ok', Result};
```
Comment (lazedo) [inline core/kazoo_data/src/kz_datamgr.erl]: please a clause for multiple results and return `{'error', 'multiple_results'}`

Comment (fenollp) [inline core/kazoo_data/src/kz_datamgr.erl]: comming up in next commit

**Comment by fenollp** [comment]
@lazedo ready

**Pull request merged**